File: source_funcs/go_transcoder/NUMBER_JUMP_REQUIRED_GIVEN_LENGTH_REACH_POINT_FORM_D_0_ORIGIN_2D_PLANE/NUMBER_JUMP_REQUIRED_GIVEN_LENGTH_REACH_POINT_FORM_D_0_ORIGIN_2D_PLANE.go
package main

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}

func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}

func f_gold(a int, b int, d int) int {
	var temp int = a
	a = min(a, b)
	b = max(temp, b)
	if d >= b {
		return (d + b - 1) / b
	}
	if d == 0 {
		return 0
	}
	if d == a {
		return 1
	}
	return 2
}

File: rust_funcs/c_transcoder/NUMBER_JUMP_REQUIRED_GIVEN_LENGTH_REACH_POINT_FORM_D_0_ORIGIN_2D_PLANE/NUMBER_JUMP_REQUIRED_GIVEN_LENGTH_REACH_POINT_FORM_D_0_ORIGIN_2D_PLANE.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}


use std::f32;

fn f_gold(mut a: i32, mut b: i32, mut d: i32) -> i32 {
    let mut temp = a;
    a = a.min(b);
    b = temp.max(b);
    if d >= b {
        return (d + b - 1) as i32 / b;
    } 
    if d == 0 {
        return 0;
    }
    if d == a  {
        return 1;
    }
    return 2;
}
